syncer/syncer.go

```diff
@@ -1826,103 +1754,14 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
-			err = s.onlineDDL.Finish(ec.tctx, table.Schema, table.Name)
-			if err != nil {
-				return terror.Annotatef(err, "finish online ddl on %s.%s", table.Schema, table.Name)
-			}
-		}
-
-		return nil
-	}
-
-	// handle shard DDL in optimistic mode.
+	// handle shard DDL
 	if s.cfg.ShardMode == config.ShardOptimistic {
 		return s.handleQueryEventOptimistic(ev, ec, needHandleDDLs, needTrackDDLs, onlineDDLTableNames)
+	} else if s.cfg.ShardMode == config.ShardPessimistic {
+		return s.handleQueryEventPessimistic(ev, ec, needHandleDDLs, needTrackDDLs, onlineDDLTableNames, ddlInfo, sqls)
 	}
 
-	// handle sharding ddl
-	var (
-		needShardingHandle bool
-		group              *ShardingGroup
-		synced             bool
-		active             bool
-		remain             int
-		source             string
-	)
-	// for sharding DDL, the firstPos should be the `Pos` of the binlog, not the `End_log_pos`
-	// so when restarting before sharding DDLs synced, this binlog can be re-sync again to trigger the TrySync
-	startLocation := ec.startLocation.Clone()
-
-	source, _ = GenTableID(ddlInfo.tableNames[0][0].Schema, ddlInfo.tableNames[0][0].Name)
-
-	var annotate string
-	switch ddlInfo.stmt.(type) {
-	case *ast.CreateDatabaseStmt:
-		// for CREATE DATABASE, we do nothing. when CREATE TABLE under this DATABASE, sharding groups will be added
-	case *ast.CreateTableStmt:
-		// for CREATE TABLE, we add it to group
-		needShardingHandle, group, synced, remain, err = s.sgk.AddGroup(ddlInfo.tableNames[1][0].Schema, ddlInfo.tableNames[1][0].Name, []string{source}, nil, true)
-		if err != nil {
-			return err
-		}
-		annotate = "add table to shard group"
-	default:
-		needShardingHandle, group, synced, active, remain, err = s.sgk.TrySync(ddlInfo.tableNames[1][0].Schema, ddlInfo.tableNames[1][0].Name, source, startLocation.Clone(), ec.currentLocation.Clone(), needHandleDDLs)
-		if err != nil {
-			return err
-		}
-		annotate = "try to sync table in shard group"
-		// meets DDL that will not be processed in sequence sharding
-		if !active {
-			s.tctx.L().Info("skip in-activeDDL", zap.String("event", "query"), zap.String("source", source), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), zap.Bool("in-sharding", needShardingHandle), zap.Stringer("start location", startLocation), zap.Bool("is-synced", synced), zap.Int("unsynced", remain))
-			return nil
-		}
-	}
-
-	s.tctx.L().Info(annotate, zap.String("event", "query"), zap.String("source", source), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), zap.Bool("in-sharding", needShardingHandle), zap.Stringer("start location", startLocation), zap.Bool("is-synced", synced), zap.Int("unsynced", remain))
+	s.tctx.L().Info("start to handle ddls in normal mode", zap.String("event", "query"), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), log.WrapStringerField("location", ec.currentLocation))
```

Comment:
this is the original logic of `if s.cfg.ShardMode == "" {}`, git diff doesn't recognize it